diff --git a/acapy_agent/admin/server.py b/acapy_agent/admin/server.py
index 93be1b76f..1f7f0e814 100644
--- a/acapy_agent/admin/server.py
+++ b/acapy_agent/admin/server.py
@@ -419,7 +419,7 @@ class AdminServer(BaseAdminServer):
         version_string = f"v{__version__}"
 
         setup_aiohttp_apispec(
-            app=app, title=agent_label, version=version_string, swagger_path="/api/doc"
+            app=app, title=agent_label, version=version_string, swagger_path="/api/doc", swagger_ui_settings={ "docExpansion": "none" }
         )
         app.on_startup.append(self.on_startup)
 
